src/foam/core/XML.js

```diff
@@ -0,0 +1,466 @@
+      if ( ! first ) this.out(',').nl().indent();
+      this.out(this.maybeEscapeKey(key), ':').output(value);
+    },
+
+    function outputObjectKeyValues_(o) {
+      var first = true;
+      for ( var key in o ) {
+        this.outputObjectKeyValue_(key, o[key], first);
+        first = false;
+      }
+    },
+
+    function outputSortedObjectKeyValues_(o) {
+      var key, keys = [];
+
+      for ( key in o ) keys.push(key);
+      keys.sort();
+
+      var first = true;
+      for ( var i = 0 ; i < keys.length; i++ ) {
+        key = keys[i];
+        this.outputObjectKeyValue_(key, o[key], first);
+        first = false;
+      }
+    },
+
+    {
+      name: 'output',
+      code: foam.mmethod({
+        Undefined: function(o) { this.out('null'); },
+        Null:      function(o) { this.out('null'); },
+        String:    function(o) { this.out(this.escape(o)); },
+        Number:    function(o) { this.out(o); },
+        Boolean:   function(o) { this.out(o); },
+        Date:      function(o) { this.outputDate(o); },
+        Function:  function(o) { this.outputFunction(o); },
+        AbstractEnum: function(o) { },
+        FObject:   function(o) {
+          if ( o.outputXML ) {
+            o.outputXML(this)
+            return;
+          }
+
+          var clsName = o.cls_.id;
+          this.start("<object class='" + clsName + "'>")
+          // Iterate through properties and output
+          var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+          for ( var i = 0 ; i < ps.length ; i++ ) {
+            this.outputProperty_(o, ps[i]);
+          }
+          this.end('</object>');
+        },
+        Array: function(o) {
+          // Nested Objects and FObject Arrays Passed
+          for ( var i = 0 ; i < o.length ; i++ ) {
+            // Output 'value' tags for arrays containing non-FObject values
+            if ( !o[i].cls_) this.out("<value>");
+            this.output(o[i], this);
+            if ( !o[i].cls_ ) this.out("</value>");
+            if ( o.length - i != 1 ) this.nl().indent();
+          }
+        },
+        Object: function(o) {
+          if ( o.outputXML ) {
+            o.outputXML(this);
+          } else {
+            var oName = o.name;
+            this.start("<object name='" + oName + "'>");
+            if ( this.sortObjectKeys ) {
+              this.outputSortedObjectKeyValues_(o);
+            } else {
+              this.outputObjectKeyValues_(o);
+            }
+            this.end('</object>');
+          }
+        }
+      })
+    },
+
+    function stringify(o) {
+      // Root tags of objects for array of FObjects
+      if ( o instanceof Array ) {
+        this.start("<objects>");
+        this.output(o);
+        this.end("</objects>");
+      } else {
+        this.output(o);
+      }
+      var ret = this.buf_;
+      this.reset(); // reset to avoid retaining garbage
+      return ret;
+    },
+
+    {
+      name: 'createFObj',
```

Comment:
Would this be better named parse rather than createFObject?

It's responsible for taking some sort of document representation and then turning it into real FObject instances, much like foam.core.JSON.parse is.

In particular, it doesn' tjust create one FObject like the name would imply, if given an array it may create a whole bunch.